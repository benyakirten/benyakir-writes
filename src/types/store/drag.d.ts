interface DragState {
  draggedIndex: number
  draggedValue: string
}
