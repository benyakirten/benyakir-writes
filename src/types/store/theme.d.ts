type ThemeState = {
	themes: BaseTheme[];
	ignoreComputerPreferences: boolean;
	active: BaseTheme;
	latestId: number;
};

type BasicThemeSection = {
	background: string;
	textColor: string;
};

type ThemeBase = BasicThemeSection & {
	border: string;
	link: string;
	pageGradient: string;
};

type ThemeSidebar = {
	gradient: string;
	shadowColor: string;
};

type ThemeTextInput = BasicThemeSection & {
	border: string;
	disabled: string;
};

type ThemeMultipleChoice = BasicThemeSection & {};

type ThemeFillIn = {
	border: string;
	default: BasicThemeSection;
	disabled: BasicThemeSection;
	hover: BasicThemeSection;
};

type ThemeToggle = {
	onColor: string;
	onBackground: string;
	offColor: string;
	offBackground: string;
	border: string;
};

type ThemeHeader = {
	startColor: string;
	endColor: string;
};

type ThemeThemePage = {
	iconColor: string;
	iconHoverColor: string;
	selectedThemeColor: string;
};

type ThemeCard = {
	arrowColor: string;
	boxShadow: string;
};

type BaseTheme = {
	id: string;
	name: string;
	base: ThemeBase;
	pill: BasicThemeSection;
	header: ThemeHeader;
	sidebar: ThemeSidebar;
	textInput: ThemeTextInput;
	multipleChoice: ThemeMultipleChoice;
	fillIn: ThemeFillIn;
	toggle: ThemeToggle;
	theme: ThemeThemePage;
	card: ThemeCard;
};

type RecursiveControlGroup = {
	[key: string]: string | RecursiveControlGroup;
};
