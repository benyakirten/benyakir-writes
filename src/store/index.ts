import { configureStore } from '@reduxjs/toolkit'
import * as Redux from 'redux';
import { logger } from 'redux-logger';

import themeReducer from './theme/theme.slice';

const otherMiddleware: Redux.Middleware[] = []

if (process.env.NODE_ENV === 'development') {
  otherMiddleware.push(logger)
}

const store =  configureStore({
  reducer: {
    theme: themeReducer
  },
  middleware: (getDefaultMiddleWare) => getDefaultMiddleWare().concat(...otherMiddleware)
})

export default store;

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;