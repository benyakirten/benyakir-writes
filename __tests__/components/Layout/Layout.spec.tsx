import React from "react";
import { Provider } from "react-redux";
import { cleanup, render } from "@testing-library/react";

import store from "@/store";
import { Layout } from "@Layout";

describe("Layout component", () => {
  jest.mock("@reach/router");

  afterEach(cleanup);

  it("should render correctly", () => {
    expect(() =>
      render(
        <Provider store={store}>
          <Layout />
        </Provider>
      )
    ).not.toThrow();
  });
});
