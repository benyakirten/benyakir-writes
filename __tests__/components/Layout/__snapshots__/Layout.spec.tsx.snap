// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should render correctly 1`] = `
<div
  className="sc-bdfBQB iRScpY"
>
  <nav
    className="sc-dlfnuX dhFKmv nav-toggle-close"
    data-navtoggle="nav-toggle"
    onClick={[Function]}
    open={false}
  >
    <div
      className="sc-hKgJUU botUjJ nav-toggle-close"
      data-navtoggle="nav-toggle"
    >
      <button
        className="sc-eCstlR gHjuob"
        onClick={[Function]}
        open={false}
        tabIndex={0}
      >
        ←
      </button>
      <div
        aria-hidden={true}
        className="sc-jSgvzq eQMidy nav-toggle-close"
        data-navtoggle="nav-toggle"
        open={false}
      >
        <div
          className="sc-gKseQn cOJtMT"
        >
          <div
            className="sc-bqyKOL vFWqF"
            onClick={[Function]}
            open={false}
          >
            <span
              className="sc-hBEYId cVqZVR"
              onClick={[Function]}
              open={false}
              role="button"
              tabIndex={-1}
            >
              ↡
            </span>
            <a
              active={false}
              className="sc-fodVek hmiHRn"
              dark={true}
              href="/blog"
              inline={false}
              limitUnderbar={false}
              small={false}
              tabIndex={-1}
            >
              BLOG
            </a>
            <div
              aria-hidden={true}
              className="sc-kstqJO brGSJA"
              data-navtoggle="no-toggle"
              height="7rem"
              open={false}
            >
              <a
                active={false}
                className="sc-fodVek dnZZub"
                dark={true}
                href="/blog/bens-thoughts/"
                inline={false}
                limitUnderbar={false}
                small={true}
                tabIndex={-1}
                underbarSize="16rem"
              >
                Ben's Thoughts
              </a>
              <a
                active={false}
                className="sc-fodVek dnZZub"
                dark={true}
                href="/blog/reviews/"
                inline={false}
                limitUnderbar={false}
                small={true}
                tabIndex={-1}
                underbarSize="16rem"
              >
                Reviews
              </a>
              <a
                active={false}
                className="sc-fodVek dnZZub"
                dark={true}
                href="/blog/down-south-boulder-road/"
                inline={false}
                limitUnderbar={false}
                small={true}
                tabIndex={-1}
                underbarSize="16rem"
              >
                Down South Boulder Road
              </a>
            </div>
          </div>
          <div
            className="sc-bqyKOL vFWqF"
            onClick={[Function]}
            open={false}
          >
            <span
              className="sc-hBEYId cVqZVR"
              onClick={[Function]}
              open={false}
              role="button"
              tabIndex={-1}
            >
              ↡
            </span>
            <a
              active={false}
              className="sc-fodVek hmiHRn"
              dark={true}
              href="/author"
              inline={false}
              limitUnderbar={false}
              small={false}
              tabIndex={-1}
            >
              AUTHOR
            </a>
            <div
              aria-hidden={true}
              className="sc-kstqJO brGSJA"
              data-navtoggle="no-toggle"
              height="4rem"
              open={false}
            >
              <a
                active={false}
                className="sc-fodVek dnZZub"
                dark={true}
                href="/author/books/"
                inline={false}
                limitUnderbar={false}
                small={true}
                tabIndex={-1}
                underbarSize="16rem"
              >
                Books
              </a>
              <a
                active={false}
                className="sc-fodVek dnZZub"
                dark={true}
                href="/author/short-stories/"
                inline={false}
                limitUnderbar={false}
                small={true}
                tabIndex={-1}
                underbarSize="16rem"
              >
                Short Stories
              </a>
            </div>
          </div>
          <a
            active={false}
            className="sc-fodVek lgTBpb"
            dark={true}
            href="/portfolio"
            inline={false}
            limitUnderbar={false}
            small={false}
            tabIndex={-1}
            underbarSize="12rem"
          >
            Portfolio
          </a>
        </div>
        <div
          className="sc-gKseQn cOJtMT"
        >
          <div
            className="sc-bqyKOL vFWqF"
            onClick={[Function]}
            open={false}
          >
            <span
              className="sc-hBEYId cVqZVR"
              onClick={[Function]}
              open={false}
              role="button"
              tabIndex={-1}
            >
              ↡
            </span>
            <h3
              className="sc-iwyWTf hpbdPn"
            >
              Search
            </h3>
            <div
              aria-hidden={true}
              className="sc-kstqJO brGSJA"
              data-navtoggle="no-toggle"
              height="auto"
              open={false}
            >
              <div
                className="sc-jgHCSr eWvmVj"
              >
                <input
                  autoFocus={false}
                  id="global-search"
                  name="global-search"
                  onChange={[Function]}
                  placeholder=" "
                  style={
                    Object {
                      "width": "65%",
                    }
                  }
                  tabIndex={-1}
                  type="text"
                  value=""
                />
                <label
                  htmlFor="global-search"
                >
                  Search
                </label>
              </div>
              <div
                style={
                  Object {
                    "marginLeft": "2px",
                    "marginTop": "1rem",
                  }
                }
              >
                <div
                  className="sc-gTgzoy bjvVbk"
                >
                  <input
                    aria-labelledby="label-global-search-show-post"
                    checked={true}
                    id="global-search-show-post"
                    name="global-search-show-post"
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <label
                    data-navtoggle="no-toggle"
                    onClick={[Function]}
                    tabIndex={-1}
                  >
                    <span
                      data-navtoggle="no-toggle"
                    >
                      ☞
                    </span>
                  </label>
                  <span
                    id="label-global-search-show-post"
                    onClick={[Function]}
                  >
                    Show Posts
                  </span>
                </div>
                <div
                  className="sc-gTgzoy bjvVbk"
                >
                  <input
                    aria-labelledby="label-global-search-show-project"
                    checked={true}
                    id="global-search-show-project"
                    name="global-search-show-project"
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <label
                    data-navtoggle="no-toggle"
                    onClick={[Function]}
                    tabIndex={-1}
                  >
                    <span
                      data-navtoggle="no-toggle"
                    >
                      ☞
                    </span>
                  </label>
                  <span
                    id="label-global-search-show-project"
                    onClick={[Function]}
                  >
                    Show Projects
                  </span>
                </div>
                <div
                  className="sc-gTgzoy bjvVbk"
                >
                  <input
                    aria-labelledby="label-global-search-show-book"
                    checked={true}
                    id="global-search-show-book"
                    name="global-search-show-book"
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <label
                    data-navtoggle="no-toggle"
                    onClick={[Function]}
                    tabIndex={-1}
                  >
                    <span
                      data-navtoggle="no-toggle"
                    >
                      ☞
                    </span>
                  </label>
                  <span
                    id="label-global-search-show-book"
                    onClick={[Function]}
                  >
                    Show Books
                  </span>
                </div>
                <div
                  className="sc-gTgzoy bjvVbk"
                >
                  <input
                    aria-labelledby="label-global-search-show-story"
                    checked={true}
                    id="global-search-show-story"
                    name="global-search-show-story"
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <label
                    data-navtoggle="no-toggle"
                    onClick={[Function]}
                    tabIndex={-1}
                  >
                    <span
                      data-navtoggle="no-toggle"
                    >
                      ☞
                    </span>
                  </label>
                  <span
                    id="label-global-search-show-story"
                    onClick={[Function]}
                  >
                    Show Stories
                  </span>
                </div>
              </div>
              <div
                className="sc-khAkCZ geYNqj"
              >
                <div
                  className="sc-giImIA fPLbSg"
                >
                  No results yet!
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-gKseQn cOJtMT"
        >
          <a
            active={false}
            className="sc-fodVek lgTBpb"
            dark={true}
            href="/"
            inline={false}
            limitUnderbar={false}
            small={false}
            tabIndex={-1}
            underbarSize="12rem"
          >
            Home
          </a>
          <a
            active={false}
            className="sc-fodVek lgTBpb"
            dark={true}
            href="/contact"
            inline={false}
            limitUnderbar={false}
            small={false}
            tabIndex={-1}
            underbarSize="12rem"
          >
            Contact
          </a>
          <a
            active={false}
            className="sc-fodVek lgTBpb"
            dark={true}
            href="/privacy"
            inline={false}
            limitUnderbar={false}
            small={false}
            tabIndex={-1}
            underbarSize="12rem"
          >
            Privacy
          </a>
          <a
            active={false}
            className="sc-fodVek lgTBpb"
            dark={true}
            href="/about"
            inline={false}
            limitUnderbar={false}
            small={false}
            tabIndex={-1}
            underbarSize="12rem"
          >
            About
          </a>
        </div>
      </div>
      <div
        className="sc-gKseQn cOJtMT"
      >
        <div
          className="sc-fubCzh hgeMBo"
          open={false}
        >
          <span
            className="sc-pGacB RvNja"
          >
            © 2021 by Benyakir Horowitz
          </span>
          <span
            className="sc-pGacB RvNja"
          >
            All Rights Reserved
          </span>
        </div>
        <a
          href="/"
        >
          <img
            alt="Benyakir Writes Logo"
            className="sc-bkzYnD iEBnrP"
          />
        </a>
      </div>
    </div>
  </nav>
  <main
    className="sc-gsTEea gEMJgI"
  >
    <div>
      <div
        style={
          Object {
            "opacity": 1,
            "transform": "translateY(0)",
            "transition": "all 500ms ease-in-out",
          }
        }
      />
    </div>
  </main>
</div>
`;
